src/main/java/org/spongepowered/api/profile/GameProfileManager.java

```diff
@@ -99,20 +157,54 @@
     ListenableFuture<Collection<GameProfile>> getAllByName(Iterable<String> names, boolean useCache);
 
     /**
-     * Gets a collection of {@link GameProfile}s by their unique IDs.
+     * Fills a {@link GameProfile}.
      *
-     * @param uniqueIds The UUIDs
+     * <p>The returned {@link ListenableFuture} throws an {@link ExecutionException}
+     * caused by a {@link ProfileNotFoundException} if the profile does not exist or
+     * an {@link IOException} if a network error occurred.</p>
+     *
+     * @param profile The profile to fill
+     * @return The result of the request
+     */
+    default ListenableFuture<GameProfile> fill(GameProfile profile) {
+        return this.fill(profile, false);
+    }
+
+    /**
+     * Fills a {@link GameProfile}.
+     *
+     * <p>The returned {@link ListenableFuture} throws an {@link ExecutionException}
+     * caused by a {@link ProfileNotFoundException} if the profile does not exist or
+     * an {@link IOException} if a network error occurred.</p>
+     *
+     * @param profile The profile to fill
+     * @param signed true if we should request that the profile data be signed
+     * @return The result of the request
+     */
+    default ListenableFuture<GameProfile> fill(GameProfile profile, boolean signed) {
+        return this.fill(profile, signed, true);
+    }
+
+    /**
+     * Fills a {@link GameProfile}.
+     *
+     * <p>The returned {@link ListenableFuture} throws an {@link ExecutionException}
+     * caused by a {@link ProfileNotFoundException} if the profile does not exist or
+     * an {@link IOException} if a network error occurred.</p>
+     *
+     * @param profile The profile to fill
+     * @param signed true if we should request that the profile data be signed
      * @param useCache true to perform a cache lookup first
      * @return The result of the request
      */
-    ListenableFuture<Collection<GameProfile>> getAllById(Iterable<UUID> uniqueIds, boolean useCache);
+    ListenableFuture<GameProfile> fill(GameProfile profile, boolean signed, boolean useCache);
```

Comment:
I don't really think so. It's perfectly possible for a plugin to construct a `GameProfile` on its own, with just a UUID (or UUID and name). There should be a way to fill it with the appropriate profile properties.